### PR TITLE
python3Packages.mechanicalsoup: 1.0.0 -> 1.1.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.1.0

Change log: https://github.com/MechanicalSoup/MechanicalSoup/releases/tag/v1.1.0

- Rename acc. PEP503
- Switch to GitHub as source
- Enable tests

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 124978` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.mechanicalsoup</li>
    <li>python39Packages.mechanicalsoup</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124978` at 54584a99 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.mechanicalsoup</li>
    <li>python39Packages.mechanicalsoup</li>
  </ul>
</details>

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -4213,7 +4213,7 @@ in {
 
   mecab-python3 = callPackage ../development/python-modules/mecab-python3 { };
 
-  MechanicalSoup = callPackage ../development/python-modules/MechanicalSoup { };
+  mechanicalsoup = callPackage ../development/python-modules/mechanicalsoup { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:4216]: Please add an alias for this to python-aliases.nix.

File: pkgs/development/python-modules/mechanicalsoup/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      --replace " --cov --cov-config .coveragerc --flake8" ""
+  '';
+
+  pythonImportsCheck = [ "mechanicalsoup" ];
+
+  meta = with lib; {
+    description = "A Python library for automating interaction with websites";
+    homepage = "https://github.com/hickford/MechanicalSoup";
+    license = licenses.mit;
+    maintainers = [ maintainers.jgillich ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mechanicalsoup/default.nix:51]: Maybe add yourself as a maintainer?

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/top-level/python-packages.nix:4216]: Done, thanks.

File: pkgs/top-level/python-aliases.nix [hunk 3]
```diff
@@ -36,4 +36,5 @@ mapAliases ({
   smart_open = smart-open; # added 2021-03-14
   google_api_python_client = google-api-python-client; # added 2021-03-19
   googleapis_common_protos = googleapis-common-protos; # added 2021-03-19
+  mechanicalsoup = MechanicalSoup;  # added 2021-06-01
```
Comment (arcnmx) [inline pkgs/top-level/python-aliases.nix:39]: Isn't this backwards, how did this ever build? It shadows the actual `pythonPackages.mechanicalsoup` attr afaict

**Pull request merged**